lib/gprdmat.gi

```diff
@@ -416,6 +416,49 @@ local  info,proj,H;
   return proj;
 end);
 
+#############################################################################
+##
+#M  ListWreathProductElementOp( <G>, <x> )
+##
+InstallMethod( ListWreathProductElementOp, "matrix wreath product", true,
+ [ IsMatrixGroup and HasWreathProductInfo, IsObject ], 0,
+function(G, x)
+  local info, degI, dimA, field, base, top, topImages, block, k, l;
+
+  info := WreathProductInfo(G);
+  degI := info.degI;
+  dimA := info.dimA;
+  field := info.field;
+
+  base := [];
+  topImages := [];
+  for k in [1 .. degI] do
+      for l in [1 .. degI] do
+          block := x{[dimA * (k - 1) + 1 .. dimA * k]}{[dimA * (l - 1) + 1 .. dimA * l]};
```

Comment:
```suggestion
          block := ExtractSubMatrix(x, [dimA * (k - 1) + 1 .. dimA * k], [dimA * (l - 1) + 1 .. dimA * l]);
```
Doing `x{rows}{cols}` first creates an intermediate object `x{rows}` and then selects the columns.

Reply:
Thanks for showing me this improvement.